### ISSUE TITLE
Default profile's cli_pager is ignored

### ISSUE THREAD

**Issue by furikake** [aws/aws-cli#5951]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone though the [User Guide](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-welcome.html) and the [API reference](https://docs.aws.amazon.com/cli/latest/reference/)
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-cli/issues) and didn't find any solution
  
**Describe the bug**
According to https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-pagination.html#cli-usage-pagination-clientside the order of precedence is:

> AWS CLI version 2 provides the use of a client-side pager program for output. By default, this feature returns all output through your operating system’s default pager program.
> 
> In order of precedence, you can specify the output pager in the following ways:
> 
> - Using the cli_pager setting in the config file in a named profile.
> - Using the AWS_PAGER environment variable.
> - Using the cli_pager setting in the config file in default profile.
> - Using the PAGER environment variable.
> 
> In order of precedence, you can disable all use of an external paging program in the following ways:
> 
> - Use the --no-cli-pager command line option to disable the pager for a single command use.
> - Set the cli_pager setting or AWS_PAGER variable to an empty string.

I think setting `cli_pager` in `default` profile is broken as named profile should without a `cli_pager` setting should inherit that from default profile.  This doesn't happen.

**SDK version number**

```
❯ aws --version
aws-cli/2.1.26 Python/3.9.1 Darwin/19.6.0 source/x86_64 prompt/off
```

**Platform/OS/Hardware/Device**

MacOS Catalina 10.15.7 on MacBook Pro 13" 2020

**To Reproduce (observed behavior)**

I would like to disable the cli_pager by default and selectively configure it for certain profiles.  An example config for repro is:

```
❯ cat ~/.aws/config
[default]
region = us-east-1
output = json
cli_pager =

[profile pager]
region = us-east-1
output = table
cli_pager = more

[profile no-pager]
region = ap-southeast-2
```

There is a `PAGER` environment variable but unsetting it doesn't change anything.  There's no `AWS_PAGER` set.

```
❯ echo "$PAGER"
less
❯ echo "$AWS_PAGER"

```

Running `aws sts get-caller-identity --profile pager` does the right thing by using `more` as pager.

I would expect that running `aws sts get-caller-identity --profile no-pager` would use the default profile's pager of <BLANK>.  Therefore, no paging.  The output is using `less`.

**Expected behavior**

Running `aws sts get-caller-identity --profile no-pager` should have not paged using `less` as it should have used the `default` profile's pager of <BLANK> (disabled).

**Logs/output**

N/A

**Additional context**

N/A

**Comment by caylorme** [comment]
I can confirm this behavior

```aws-cli/2.1.17 Python/3.7.4 Darwin/19.6.0 exe/x86_64 prompt/off```

setting the value to null in default section as described in the documentation fails to remove pagination https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-pagination.html#cli-usage-pagination-clientside

Using the `AWS_PAGER` environment variable works, as does adding the parameter to named profile sections.

**Comment by kdaily** [comment]
Hi @furikake and @caylorme,

The documentation for this is not clear. The pager configuration works with named profiles just like any other configurable parameter. The default profile is not used as a fall back - it is only used if you do not specify a profile (either with a command line parameter or environment variable), or if you explicitly set the profile to be default.

You can read about the configuration precedence options here:

https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-precedence

To get the behavior you want, you would need to explicitly set the pager behavior for each profile. 

There are a couple of open issues that relate to inheriting parameters from other profiles. It seems this is the behavior you want or are expecting, but this is not available now: (example: #4800)

I will do two things:

1. submit a documentation issue to clarify this behavior, and 
2. open a tracking issue for parameter inheritance and merge other requests there.

Thanks for bringing it to our attention!
